### PR TITLE
Changes needed for substrate#4612

### PR DESCRIPTION
**Pull request by expenses**
See https://github.com/paritytech/substrate/pull/4612. This merges in https://github.com/paritytech/finality-grandpa/pull/100, but also:

* Changes `rand` to be dependent on the `test-helpers` feature, which means that the crate actually compiles with this feature turned on. This does mean that you have to run `cargo test --feature test-helpers` to test it though.
* Adds an `Unpin` impl for a few types.

### CODE REVIEW

**Comment by rphmeier** [comment]
> Changes rand to be dependent on the test-helpers feature, which means that the crate actually compiles with this feature turned on. This does mean that you have to run cargo test --feature test-helpers to test it though.

That is pretty unergonomic. Can you explain the rationale?

**Comment by expenses** [comment]
@rphmeier I've added rand back as a `dev-dependency` now, so it should just work.

File: Cargo.toml [hunk 1]
```diff
@@ -8,20 +8,20 @@ repository = "https://github.com/paritytech/finality-grandpa"
 edition = "2018"
 
 [dependencies]
-futures = "0.1"
+futures = "0.3.1"
+futures-timer = "2.0.2"
```
Comment (rphmeier) [inline Cargo.toml]: does it still compile in `no-std`? I would have thought we needed `no-default-features` on these futures crates.

**Comment by expenses** [comment]
@rphmeier I need to test that.

File: Cargo.toml (hunk 1 above)
Comment (expenses) [reply Cargo.toml]: Fixed.

**Comment by expenses** [comment]
@rphmeier I think we're good to go here. Might be best to merge in https://github.com/paritytech/finality-grandpa/pull/100 first if you're happy with that.

**Comment by rphmeier** [comment]
Solve conflicts, then merge here. Thanks!

**Comment by rphmeier** [comment]
We'll need to publish a new GRANDPA version; cc @andresilva

**Pull request merged**